pkgs/development/tools/build-managers/cmake/setup-hook.sh

```diff
@@ -50,13 +50,12 @@ cmakeConfigurePhase() {
     # because we usually do not package the framework
     cmakeFlags="-DCMAKE_FIND_FRAMEWORK=last $cmakeFlags"
 
+    # on macOS i686 was only relevant for 10.5 or earlier.
+    cmakeFlags="-DCMAKE_OSX_ARCHITECTURES=x86_64 $cmakeFlags"
+
     # we never want to use the global macOS SDK
     cmakeFlags="-DCMAKE_OSX_SYSROOT= $cmakeFlags"
 
-    # disable OSX deployment target
-    # we don't want our binaries to have a "minimum" OSX version
-    cmakeFlags="-DCMAKE_OSX_DEPLOYMENT_TARGET= $cmakeFlags"
```

Comment:
I'm surprised this isn't needed? How does CMake choose this value when it's unset?

Reply:
> If not set explicitly the value is initialized by the MACOSX_DEPLOYMENT_TARGET environment variable, if set, and otherwise computed based on the host platform.

So if I understand this correctly this was equivalent to unsetting MACOSX_DEPLOYMENT_TARGET from the stdenv.